### PR TITLE
[tests-only][full-ci]add edge case for check file info

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR add tests for edge cases
Screenshot from 2024-09-18 13-08-08

## Related Issue

- https://github.com/owncloud/ocis/issues/9844

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 1]
```diff
@@ -228,4 +246,53 @@ public function userTriesToCreateAFileInsideDeletedFolderUsingWopiEndpoint(strin
...
+	/**
+	 * @When user :user tries to get the file information of file using wopi endpoint
+	 * @When user :user gets the file information of file using wopi endpoint
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userTriesToCheckTheInformationOfDeletedFileUsingWopiEndpoint(string $user):void {
```
Comment (S-Panta) [inline tests/acceptance/bootstrap/CollaborationContext.php:282]:
```suggestion
	public function userGetsTheFileInformationUsingWopiEndpoint(string $user):void {
```
or something like this

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 2]
```diff
@@ -599,3 +599,620 @@ Feature: check file info with different wopi apps
...
+
+  Scenario Outline: try to get file info of deleted file with office suites
+    Given user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And user "Alice" has sent POST request on app endpoint:
+      | resource        | textfile0.txt   |
+      | space           | Personal        |
+      | suites          | <office-suites> |
+    And user "Alice" has deleted file "/textfile0.txt"
+    When user "Alice" tries to get the file information of file using wopi endpoint
+    Then the HTTP status code should be "500"
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:612]: We can check the JSON response if it returns any response.

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 3]
```diff
@@ -599,3 +599,620 @@ Feature: check file info with different wopi apps
       | Collabora     |
       | FakeOffice    |
       | OnlyOffice    |
+
+
+  Scenario Outline: try to get file info of deleted file with office suites
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:604]: Please, add the issue tag and let's wait for the fix. I see that the fix PR is ready

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 4]
```diff
@@ -599,3 +599,620 @@ Feature: check file info with different wopi apps
...
+    Given user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And user "Alice" has sent POST request on app endpoint:
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:606]: ?
```suggestion
    And user "Alice" has sent the following app-open request:
```

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature (hunk 2 above)
Comment (saw-jan) [reply tests/acceptance/features/apiCollaboration/checkFileInfo.feature:612]: looks like the 500 will be fixed soon

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 5]
```diff
@@ -599,3 +599,620 @@ Feature: check file info with different wopi apps
...
+      | resource        | textfile0.txt   |
+      | space           | Personal        |
+      | suites          | <office-suites> |
+    And user "Alice" has deleted file "/textfile0.txt"
+    When user "Alice" tries to get the file information of file using wopi endpoint
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:611]: or something that refers to previously opened file
```suggestion
    When user "Alice" tries to get the information of the last opened file using wopi endpoint
```

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature (hunk 2 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiCollaboration/checkFileInfo.feature]: > We can check the JSON response if it returns any response.

No JSON response was found so only http status code was checked. Same for 404 status code

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 6]
```diff
@@ -255,4 +273,53 @@ public function userTriesToCreateAFileInsideDeletedFolderUsingWopiEndpoint(strin
...
+	/**
+	 * @Given user :user has sent the following app-open request:
+	 *
+	 * @param string $user
+	 * @param TableNode $items
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userHasSentPostRequestOnAppEndpoint(string $user, TableNode $items): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:286]:
```suggestion
	 * @param TableNode $properties
	 *
	 * @return void
	 * @throws GuzzleException
	 */
	public function userHasSentTheFollowingAppOpenRequest(string $user, TableNode $properties): void {
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 7]
```diff
@@ -33,6 +34,7 @@
 class CollaborationContext implements Context {
 	private FeatureContext $featureContext;
 	private SpacesContext $spacesContext;
+	private string $storeLastAppEndpointResponse;
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:37]:
```suggestion
	private string $lastAppOpenResponse;
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 8]
```diff
@@ -52,6 +54,22 @@ public function before(BeforeScenarioScope $scope): void {
 		$this->spacesContext = $environment->getContext('SpacesContext');
 	}
 
+	/**
+	 * @param string $appData
+	 *
+	 * @return void
+	 */
+	public function storeLastAppEndpointResponse(string $appData): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:62]:
```suggestion
	public function setLastAppOpenResponse(string $appData): void {
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 9]
```diff
@@ -52,6 +54,22 @@ public function before(BeforeScenarioScope $scope): void {
...
+		$this->storeLastAppEndpointResponse = $appData;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getLastAppEndpointResponse(): string {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:69]:
```suggestion
	public function getLastAppOpenResponse(): string {
```

File: tests/acceptance/features/apiCollaboration/checkFileInfo.feature [hunk 10]
```diff
@@ -599,3 +599,620 @@ Feature: check file info with different wopi apps
...
       | FakeOffice    |
       | OnlyOffice    |
+
+  @issue-10097
+  Scenario Outline: try to get file info of deleted file with office suites
+    Given user "Alice" has uploaded file with content "hello world" to "/textfile0.txt"
+    And user "Alice" has sent the following app-open request:
+      | resource        | textfile0.txt   |
+      | space           | Personal        |
+      | suites          | <office-suites> |
```
Comment (saw-jan) [inline tests/acceptance/features/apiCollaboration/checkFileInfo.feature:609]:
```suggestion
      | app          | <office-suites> |
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 11]
```diff
@@ -33,6 +34,7 @@
 class CollaborationContext implements Context {
 	private FeatureContext $featureContext;
 	private SpacesContext $spacesContext;
+	private string $lastAppOpenResponse;
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:37]: it is actually not a reponse but a response body, so this would be good name
```suggestion
	private string $lastAppOpenData;
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 12]
```diff
@@ -52,6 +54,22 @@ public function before(BeforeScenarioScope $scope): void {
 		$this->spacesContext = $environment->getContext('SpacesContext');
 	}
 
+	/**
+	 * @param string $appData
+	 *
+	 * @return void
+	 */
+	public function setLastAppOpenResponse(string $appData): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:62]:
```suggestion
	 * @param string $data
	 *
	 * @return void
	 */
	public function setLastAppOpenData(string $data): void {
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 13]
```diff
@@ -52,6 +54,22 @@ public function before(BeforeScenarioScope $scope): void {
...
+		$this->lastAppOpenResponse = $appData;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getLastAppOpenResponse(): string {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:69]:
```suggestion
	public function getLastAppOpenData(): string {
```

File: tests/acceptance/bootstrap/CollaborationContext.php [hunk 14]
```diff
@@ -255,4 +273,53 @@ public function userTriesToCreateAFileInsideDeletedFolderUsingWopiEndpoint(strin
...
+	/**
+	 * @When user :user tries to get the information of the last opened file using wopi endpoint
+	 * @When user :user gets the information of the last opened file using wopi endpoint
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(string $user):void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/CollaborationContext.php:309]:
```suggestion
	public function userGetsTheInformationOfTheLastOpenedFileUsingWopiEndpoint(string $user): void {
```

**Review by saw-jan — APPROVED** [review]
